test/test.kafkaClient.js

```diff
@@ -235,6 +235,19 @@ describe('Kafka Client', function () {
       sandbox.clock.tick();
       sinon.assert.notCalled(client.refreshBrokerMetadata);
     });
+
+    it('should reconnect when broker is closed due to being idle', function () {
+      const client = new Client({ autoConnect: false, reconnectOnIdle: true });
+      const brokerWrapper = client.createBroker('fakehost', 9092, true);
+
+      sandbox.stub(brokerWrapper, 'isIdle').returns(true);
+      sandbox.stub(client, 'reconnectBroker');
```

Comment:
Should also stub `isValidBroker`

Reply:
I don't see `isValidBroker` anywhere in the code base, not sure what you're looking for here.

Reply:
https://github.com/SOHU-Co/kafka-node/blob/master/lib/kafkaClient.js#L831

isValidBroker is returning false that's why the travis tests are failing.

Reply:
Ahh, thx for the line ref link - my forked repo/branch wasn't synced :-( I got it now